### ISSUE TITLE
Remotes added via SSH do not work.

### ISSUE THREAD

**Issue by askvortsov1** [microsoft/vscode-pull-request-github#1656]
Remotes using ssh (git@github.com:USER/REPO) are not found, and if only they exist, an error message of 'no remotes match the existing setting' exist.

- Extension version: Nightly, 2020.4.25381 (also present on non-nightly) 
- VSCode Version: 1.45.0-insider (2020-04-15 version)
- OS: Fedora 31

Steps to Reproduce:

1. Add a repository where all remotes are added via HTTPS
2. See the 'no remotes match the existing setting' error message show up.

**Comment by RMacfarlane** [comment]
@askvortsov1 I'm having trouble reproducing this, I've tried setting remotes to ssh form and they're correctly found

```
~/dev/vscode-pull-request-github$ git remote -v
origin  git@github.com:microsoft/vscode-pull-request-github (fetch)
origin  git@github.com:microsoft/vscode-pull-request-github (push)
```

Are you able to share the full remote url you're using?

**Comment by askvortsov1** [comment]
Thanks for the quick reply! I currently have the following remotes:
origin  git@github.com:flarum/core.git (fetch)
origin  git@github.com:flarum/core.git (push)
rem     git@github.com:askvortsov1/core.git (fetch)
rem     git@github.com:askvortsov1/core.git (push)

In recent versions of nightly, the situation has changed a bit: instead of just saying that there's no matching remotes found, there's an error message in the bottom right corner saying 'The remote 'rem' is not a valid Github repository'. A similar message occasionally pops up for origin. My current list of remotes in settings is:
- origin
- rem
- upstream
- central

**Comment by RMacfarlane** [comment]
What version of the nightly extension are you using? Does the pull request tree load despite the error message? Can you open the output panel and choose "GitHub Pull Request" and copy the logs from that here?

**Comment by askvortsov1** [comment]
I'm on 2020.4.34022. The pull request tree does not load. After further testing, it looks like that error message only shows up if I add another remote over https (in which case the pull request tree DOES load), otherwise, it behaves like described in https://github.com/microsoft/vscode-pull-request-github/issues/1656#issue-603412513. Apologies for not clarifying that in my previous comment.

After examination of the output logs (forgot that existed, I was trying to look in the dev tools section), it looks like it might be an issue with my OpenSSL installation, so perhaps the solution here is documenting requirements for that / having an error message for that particular error? I put the logs below.

### Without a working https remote:
No error message in the bottom right corner shows up. However, the pull request tree is empty, and shows a 'no remotes match the current setting' message.

Remote List:
```
origin  git@github.com:flarum/core.git (fetch)
origin  git@github.com:flarum/core.git (push)
rem     git@github.com:askvortsov1/core.git (fetch)
rem     git@github.com:askvortsov1/core.git (push)
```

Output Log:
```
[Info] Review> Queuing additional validate state
[Info] Review> Validating state...
[Info] No remote with name 'origin' found. Please update your 'githubPullRequests.remotes' setting.
[Info] No remote with name 'upstream' found. Please update your 'githubPullRequests.remotes' setting.
[Info] No remote with name 'rem' found. Please update your 'githubPullRequests.remotes' setting.
[Info] No remote with name 'central' found. Please update your 'githubPullRequests.remotes' setting.
[Info] No remote with name 'https_central' found. Please update your 'githubPullRequests.remotes' setting.
[Info] No GitHub remotes found
[Info] Review> no matching pull request metadata found for current branch master
```

### With a working https remote:
An error message in the bottom right corner ocasionally shows up.

Both the PR tree and the issues tree are populated correctly.

Remote List:
```
https_central   https://github.com/flarum/core.git (fetch)
https_central   https://github.com/flarum/core.git (push)
origin  git@github.com:flarum/core.git (fetch)
origin  git@github.com:flarum/core.git (push)
rem     git@github.com:askvortsov1/core.git (fetch)
rem     git@github.com:askvortsov1/core.git (push)
```

Output Log:
```
[Info] GitHubServer> No response from host https://ssh.github.com/flarum/core: write EPROTO 68236703676168:error:100000f7:SSL routines:OPENSSL_internal:WRONG_VERSION_NUMBER:../../third_party/boringssl/src/ssl/tls_record.cc:242:

[Info] GitHubServer> No response from host https://ssh.github.com/askvortsov1/core: write EPROTO 68236703676168:error:100000f7:SSL routines:OPENSSL_internal:WRONG_VERSION_NUMBER:../../third_party/boringssl/src/ssl/tls_record.cc:242:

[Info] No remote with name 'origin' found. Please update your 'githubPullRequests.remotes' setting.
[Info] No remote with name 'upstream' found. Please update your 'githubPullRequests.remotes' setting.
[Info] No remote with name 'rem' found. Please update your 'githubPullRequests.remotes' setting.
[Info] No remote with name 'central' found. Please update your 'githubPullRequests.remotes' setting.
[Info] Found GitHub remote
[Info] Review> no matching pull request metadata found for current branch as/view_namespace_extender
```

**Comment by ryanmahan** [comment]
Were you able to solve the SSL issue? I am receiving similar logs from an enterprise github server, running 0.16.0

`[Info] serverAuthPromises failed: request to http://{serveraddress} failed, reason: write EPROTO 140648442604280:error:100000f7:SSL routines:OPENSSL_internal:WRONG_VERSION_NUMBER:../../third_party/boringssl/src/ssl/tls_record.cc:242:`

**Comment by askvortsov1** [comment]
Unfortunately no, I'm no longer sure it's an issue with my installation, I think it might be with the extension itself

**Comment by askvortsov1** [comment]
@RMacfarlane any updates on this? Thanks!

**Comment by balauppaloori** [comment]
I am also having the same issue: HTTPS remote and this extension says no remotes match the current settings

**Comment by willryan** [comment]
I'm on VSCode 1.47.3, node 12.8.1, GitHub Issues 0.18.0. I am getting this same message whenever I try to use SSH remotes:
```
[Info] serverAuthPromises failed: request to http://{serveraddress} failed, reason: write EPROTO 140648442604280:error:100000f7:SSL routines:OPENSSL_internal:WRONG_VERSION_NUMBER:../../third_party/boringssl/src/ssl/tls_record.cc:242:
```
This was true on Linux (Pop_OS) as well as WSL. Haven't tried other OSes yet.

HTTPS seems to work okay, but I much prefer to use SSH.

**Comment by ifiokjr** [comment]
I'm having the same issue.

```
Version: 1.48.0
Commit: db40434f562994116e5b21c24015a2e40b2504e6
Date: 2020-08-13T08:03:27.059Z
Electron: 7.3.2
Chrome: 78.0.3904.130
Node.js: 12.8.1
V8: 7.8.279.23-electron.0
OS: Darwin x64 19.2.0
```

And here's the error.

```
[Info] GitHubServer> No response from host https://ssh.github.com/remirror/remirror: Client network socket disconnected before secure TLS connection was established
[Info] No remote with name 'origin' found. Please update your 'githubPullRequests.remotes' setting.
[Info] No remote with name 'upstream' found. Please update your 'githubPullRequests.remotes' setting.
[Info] No GitHub remotes found
```

**Comment by leetrout** [comment]
Same issue here on Mac:

```
Version: 1.48.2
Commit: a0479759d6e9ea56afa657e454193f72aef85bd0
Date: 2020-08-25T10:09:08.021Z
Electron: 7.3.2
Chrome: 78.0.3904.130
Node.js: 12.8.1
V8: 7.8.279.23-electron.0
OS: Darwin x64 18.7.0
```

`git remote -v`:

```
origin	git@github.com:nextmv-io/info (fetch)
origin	git@github.com:nextmv-io/info (push)
```

```
Info] Looking for git repository
[Info] Git repository found, initializing review manager and pr tree view.
[Info] Review> Validate state in progress
[Info] Review> Validating state...
[Info] GitHubServer> No response from host https://ssh.github.com/nextmv-io/info: Client network socket disconnected before secure TLS connection was established
[Info] No remote with name 'origin' found. Please update your 'githubPullRequests.remotes' setting.
[Info] No GitHub remotes found
[Info] Review> no matching pull request metadata found for current branch develop
```

**Comment by GrantBlack** [comment]
pretty much the same here as well. Pretty frustrating as I am just getting started with Github and VSCode and it is dead on arrival. Just can't use the repository at all so, having to reconsider where to go. Command line tools?  

```
[Info] Git repository found, initializing review manager and pr tree view.
[Info] Review> Validate state in progress
[Info] Review> Validating state...
[Info] No remote with name 'origin' found. Please update your 'githubPullRequests.remotes' setting.
```

Pretty clearly though there is a valid remote:

```
> grant@grant-VirtualBox:~$ git remote -v
> origin	https://github.com/GrantBlack/portal.git (fetch)
> origin	https://github.com/GrantBlack/portal.git (push)
```

![image](https://user-images.githubusercontent.com/57604633/92539781-9d855a80-f296-11ea-892e-82f2b3b436a2.png)

**Comment by nperez0111** [comment]
Any progress on this? I have the same issue and it looks like pretty core functionality for this extension to have

**Comment by tomryanx** [comment]
Having a similar issue to @GrantBlack here.

```
[Debug 133281s 932ms] GitHubServer> No response from host https://192.168.199.2/<org>/<repo>: connect ETIMEDOUT 192.168.199.2:443
[Debug 133281s 932ms] GitHubServer> Host https://192.168.199.2/<org>/<repo> is associated with GitHub: false
[Debug 133281s 933ms] No remote with name 'origin' found. Please update your 'githubPullRequests.remotes' setting.
[Debug 133281s 933ms] No remote with name 'upstream' found. Please update your 'githubPullRequests.remotes' setting.
[Debug 133281s 933ms] FolderRepositoryManager> Displaying configured remotes: origin, upstream
[Debug 133281s 938ms] No GitHub remotes found
[Debug 133281s 953ms] Review> no matching pull request metadata found for current branch main
```

That IP address is on a machine-local network only used for running VMs... which I sometimes SSH in to... 

My workaround was:
```bash
$ mv ~/.ssh/config ~/.ssh/config.bk
```

I haven't yet looked into exactly /why/ vscode is getting confused by my (somewhat lengthy) ssh user config, but it's now Working(TM)[^1]
Maybe I'll post here again if I figure out what git + vscode + ssh are up to. Hope this helps someone!

[^1] after getting the non-OSS version installed and enabling the right API, but that's documented elsewhere

**Comment by azu** [comment]
Probably, I met same issue on creating issue via `Create Issue from Selection` menu.

```
There is no remote. Can't create an issue.
```

My origin is setup like this: `git remote -v`

```
$ git remote -v
origin	ssh://git@github.com/asciidwango/js-primer.git (fetch)
origin	ssh://git@github.com/asciidwango/js-primer.git (push)
```

Workaround: I've changed remote to `https://` and This extension works.

```
$ git remote -v
origin	https://github.com/asciidwango/js-primer.git (fetch)
origin	https://github.com/asciidwango/js-primer.git (push)
```

env:

- VSCode Version: 1.51.1
- Extension Version: 0.21.2

**Comment by RMacfarlane** [comment]
@azu Is there anything special in your ssh config? If you change the remotes back to how they were originally, can you check what's in the Output Panel when you select "GitHub Pull Request" in the dropdown there?

**Comment by azu** [comment]
Sorry for the delay.

Does vscode github pull request try to connect ssh.github.com?


```
[Info] Looking for git repository
[Info] Git repository found, initializing review manager and pr tree view.
[Info] Review> Validate state in progress
[Info] Review> Validating state...
[Info] GitHubServer> No response from host https://ssh.github.com/<owner>/<repo>: Client network socket disconnected before secure TLS connection was established
[Info] No remote with name 'origin' found.
[Info] No remote with name 'upstream' found.
[Info] No GitHub remotes found
[Info] Review> no matching pull request metadata found for current branch test
[Info] Review> no matching pull request metadata found on GitHub for current branch test
```

My .ssh/config

```
$ cat ~/.ssh/config

Host github.com
    User git
    Hostname ssh.github.com
    Port 443
    IdentityFile ~/.ssh/github/id_rsa.github.com
    TCPKeepAlive yes
    IdentitiesOnly yes
Host gist.github.com
    User git
    Hostname ssh.github.com
    Port 443
    IdentityFile ~/.ssh/github/id_rsa.github.com
    TCPKeepAlive yes
    IdentitiesOnly yes
```

**Comment by akshay-sf** [comment]
The same happens in my env too.
I did clone my project using `HTTPS` and `SSH`. The extension works fine with the project cloned with `HTTPS`, but it failed to load anything with the project cloned with `SSH` (no remote matches the current setting).

env:
vs-code: 1.53.2
Extension version: 0.23.1

**Comment by A-UNDERSCORE-D** [comment]
Adding a me too here, I have an upstream as a http remote, and an origin as an ssh remote. attempting to create a PR complains that it cant find a github repo for origin

vscode 1.54.3
extension ver 0.24.0

**Comment by zillionare** [comment]
not work either.

My config:
(omicron) root@150d7868b97c:/apps/omicron# git remote -v
origin  git@github.com:zillionare/omicron.git (fetch)
origin  git@github.com:zillionare/omicron.git (push)

![Uploading 图片.png…]()

**Comment by zillionare** [comment]
more log:

[Info] Looking for git repository
[Info] Git repository found, initializing review manager and pr tree view.
[Info] Review> Validate state in progress
[Info] Review> Validating state...
[Info] GitHubServer> No response from host https://ssh.github.com/zillionare/omicron: request to https://ssh.github.com/api/v3/rate_limit failed, reason: write EPROTO 140619374663552:error:1408F10B:SSL routines:ssl3_get_record:wrong version number:../deps/openssl/openssl/ssl/record/ssl3_record.c:332:

[Info] No remote with name 'origin' found.
[Info] No GitHub remotes found
[Info] Review> no matching pull request metadata found for current branch release
[Info] Review> no matching pull request metadata found on GitHub for current branch release

**Comment by zillionare** [comment]
after setting https as remote url, the extension works. But ssh is prefered. Please have this fixed, thanks!

**Comment by zillionare** [comment]
This can be work around using one url for fetch, one for push. The following config in .git/config works:
```
[remote "origin"]
        url = https://github.com/zillionare/cookiecutter-pypackage.git
        pushurl = git@github.com:zillionare/cookiecutter-pypackage.git
        fetch = +refs/heads/*:refs/remotes/origin/*
```

This allows extension pull request with https, while we can still sumbit changes by ssh.

**Comment by piotr-zdunek** [comment]
Hello All!
I'm also facing this issue. 
OS: Windows 10 Pro
Vscode version 1.55.2
Extension version: 0.25.1

PR worked when remotes were set to https. They do not work on SSH. 

Attaching log. 
[vscode_ext_log.txt](https://github.com/microsoft/vscode-pull-request-github/files/6343011/vscode_ext_log.txt)

**Comment by msgongora** [comment]
@RMacfarlane 
Here is what I have found so far since I started to face this issue 2 days ago when I updated my `.ssh/config` to use this:
```
Host github.com
    Hostname ssh.github.com
    Port 443
    User git
    IdentityFile /Users/__USERNAME__/.ssh/id_rsa-office
    IdentitiesOnly yes
```
The reason of this change is that GitHub is giving me lately, too much timeouts on port 22 with default config. This change is based on this [doc from GH](https://docs.github.com/en/github/authenticating-to-github/troubleshooting-ssh/using-ssh-over-the-https-port)

More details:

```
[Debug 1621523962.124s] Looking for git repository
[Debug 1621523962.125s] Git repository found, initializing review manager and pr tree view.
[Debug 1621523962.129s] Review> Validate state in progress
[Debug 1621523962.129s] Review> Validating state...
[Debug 1621523963.89s] GitHubServer> No response from host https://ssh.github.com/__REDACTED_ORG__/__PRIVATE_REPO__: request to https://ssh.github.com/api/v3/rate_limit failed, reason: Client network socket disconnected before secure TLS connection was established
[Debug 1621523963.89s] GitHubServer> Host https://ssh.github.com/__REDACTED_ORG__/__PRIVATE_REPO__ is associated with GitHub: false
[Debug 1621523963.891s] No remote with name 'origin' found.
[Debug 1621523963.891s] No remote with name 'upstream' found.
[Debug 1621523963.891s] FolderRepositoryManager> Displaying configured remotes: origin, upstream
[Debug 1621523964.596s] No GitHub remotes found
[Debug 1621523964.611s] Review> no matching pull request metadata found for current branch master
[Debug 1621523964.611s] Review> no matching pull request metadata found on GitHub for current branch master
```

```
git remote -v
origin  git@github.com:__REDACTED_ORG__/__PRIVATE_REPO__.git (fetch)
origin  git@github.com:__REDACTED_ORG__/__PRIVATE_REPO__.git (push)
```

**Comment by msgongora** [comment]
From the debug log I would assume that the module implementation is using the translated hostname instead of using the one set on the remote. Because of that, the module is querying an API endpoint through a Load Balancer (https://ssh.github.com) that it not meant to route traffic to API servers.

**Comment by mexin** [comment]
I had it working on OSX 11.5 but for some reason it stopped, tried reinstalling but to no avail, getting the same message
```
 GitHubServer> No response from host https://ssh.github.com/company/repository: request to https://ssh.github.com/api/v3/rate_limit failed, reason: Client network socket disconnected before secure TLS connection was established
```
Got the https over ssh from this https://docs.github.com/en/github/authenticating-to-github/troubleshooting-ssh/using-ssh-over-the-https-port working, but still, same error, cannot use https due work restrictions enforcing ssh

Any clue? 

Thank you!

**Comment by alexr00** [comment]
Should be fixed by f837b766033fd48ef28b16abcfc3f85303992f90.
